test/Chainweb/Test/Pact/RemotePactTest.hs

```diff
@@ -279,21 +278,94 @@ localChainDataTest iot nio = do
 localPreflightSimTest :: IO (Time Micros) -> IO ChainwebNetwork -> TestTree
 localPreflightSimTest iot nio = testCaseSteps "local preflight sim test" $ \step -> do
     cenv <- _getServiceClientEnv <$> nio
-    mv <- newMVar 0
+    mv <- newMVar (0 :: Int)
     sid <- mkChainId v maxBound (0 :: Int)
-
-    let pid = Pact.ChainId $ chainIdToText sid
-    step "execute preflight /local tx"
-    SubmitBatch (cmd NEL.:| []) <- testBatch'' pid iot ttl mv gp
-    res <- runLocalPreflightClient sid cenv cmd
-    checkCr res
+    let sigs = [mkSigner' sender00 []]
+
+    step "Execute preflight /local tx - known success"
+    cb0 <- mkCmdBuilder sigs v sid 1000 gp
+    SubmitBatch (cmd0 NEL.:| []) <- mkTx mv cb0
+    runLocalPreflightClient sid cenv cmd0 >>= \case
+      Left e -> assertFailure $ show e
+      Right{} -> pure ()
+
+    step "Execute preflight /local tx - unparseable chain id"
+    SubmitBatch (cmd1 NEL.:| []) <- mkRawTx mv $ Pact.ChainId "fail"
+    runClientFailureAssertion sid cenv cmd1 "Unparseable transaction chain id"
+
+    step "Execute preflight /local tx - chain id mismatch"
+    let fcid = unsafeChainId (-1)
+    SubmitBatch (cmd2 NEL.:| []) <- mkTx mv =<<
+      mkCmdBuilder sigs v fcid 1000 gp
+    runClientFailureAssertion sid cenv cmd2 "Chain id mismatch"
+
+    step "Execute preflight /local tx - tx gas limit too high"
+    SubmitBatch (cmd3 NEL.:| []) <- mkTx mv =<<
+      mkCmdBuilder sigs v sid 100000000000000 gp
+    runClientFailureAssertion sid cenv cmd3
+      "Transaction Gas limit exceeds block gas limit"
+
+    step "Execute preflight /local tx - tx gas price precision too high"
+    SubmitBatch (cmd4 NEL.:| []) <- mkTx mv =<<
+      mkCmdBuilder sigs v sid 1000 0.00000000000000001
+    runClientFailureAssertion sid cenv cmd4
+      "Gas price decimal precision too high"
+
+    step "Execute preflight /local tx - network id mismatch"
+    SubmitBatch (cmd5 NEL.:| []) <- mkTx mv =<<
+      mkCmdBuilder sigs Mainnet01 sid 1000 gp
+    runClientFailureAssertion sid cenv cmd5 "Network id mismatch"
+
+    step "Execute preflight /local tx - invalid signers/usersigs"
+
+    step "Execute preflight /local tx - too many sigs"
```

Comment:
I'm guessing this is still to be done?

Reply:
This needs to be removed because we cannot actually hit the mentioned `invalid signers/usersigs` problem via the /local endpoint (it calls `verifyCommand` which validates this)

Reply:
Got it, thanks!

Reply:
Btw, are you on an old commit? I don't see this in the diff.